Kmeans doc update (#2317)
oneapi-src/oneDAL: docs/source/onedal/algorithms/clustering/kmeans.rst

.. math::
   c_j^{(t + 1)} = \frac{1}{|S_j^{(t)}|} \sum_{x \in S_j^{(t)}} x, \quad 1 \leq j \leq k.

> If it turned out that some of :math:`S_j^{(t)}` are empty then start empty clusters handling procedure.

Reviewer: ```suggestion
If some of :math:`S_j^{(t)}` are empty, start the empty clusters handling procedure.
```

Reviewer: Not clear what is meant by "some". If even one empty thing triggers the procedure, it is better to use "any". If it is more than one, we should clarify what the minimal condition is.

Author: done

> In this case set :math:`c_j^{(t + 1)}` equal to the farthest point from just calculated centroids for each empty cluster.

Reviewer: Hmm maybe something like:

In this case, you can set the value of :math:c_j^{(t+1)} as the farthest point from the previously calculated centroids for each empty cluster. It fills in the empty clusters with new data points and makes sure that all clusters have a representative centroid.

Author: I'm not really sure that it makes sense to write 'fill' here because we are not filling the new clusters at this step. 
If you want to highlight the purpose of the procedure then what about this:

In this case, you can set the value of :math:c_j^{(t+1)} as the farthest point from the previously calculated centroids for each empty cluster. This procedure makes sure that the number of cluster remains the same.

Reviewer: I like your suggestion!

Author: done

Refer to :ref:`API Reference: K-Means <api_kmeans>`.

----------------
Distributed mode
----------------

> The algorithm supports distributed execution in SPMD mode (only on GPU).

Reviewer: ```suggestion
The algorithm supports distributed execution in the SPMD mode and can be run on a GPU only.
```

Author: Moreover, I have copied this sentence from other section which has been approved. It is necessary to change there too if something is incorrect

Author: I don't think so. The bracketed statement applies to the sentence before it and this sentence declares only that the algorithm can be run in distributed SPMD mode (as far as I understand distributed and SPMD are synonyms in this context). So, the only conclusion can be drawn is that the distributed mode is supported only for GPU.

Your last suggestion is also correct, and if we are trying to decrease brackets number in text or something like that then I can push your version :) But as for me, the current version looks more briefly and neatly.
